src/Select.js

```diff
@@ -596,12 +651,39 @@ class Select extends React.Component {
 		event.stopPropagation();
 		event.preventDefault();
 		this.setValue(this.getResetValue());
+		this.clearAutofilled();
 		this.setState({
 			isOpen: false,
 			inputValue: this.handleInputValueChange(''),
 		}, this.focus);
 	}
 
+	setAutofill () {
+		const hasValue = this.getValueArray(this.props.value).length;
```

Comment:
Why is this line here? `hasValue` isn't used, `getValueArray` doesn't seems to have any side effects.